### PR TITLE
1484 dashboard visualizations

### PR DESCRIPTION
**Pull request by jonathanbobel**
# Description
Total Allowance and Activity Bar Chart Dashboard Visualizations

## TODO (optional)

- [x] The user should be able to view a stacked bar chart visualization that represents Total Messages
- [x] The user should be able to use a dropdown to select the service or their personal account and see the Activity Snapshot. The dropdown select triggers a stacked bar chart visualization of the last 7 days of sent and failed messages.

### CODE REVIEW

File: app/templates/old/admin_template.html [hunk 1]
```diff
@@ -24,6 +24,7 @@
     <meta property="og:image" content="{{ asset_url('images/notify-og-image.png') }}">
   {% endblock %}
   <script type="text/javascript" src="{{ asset_url('js/gtm_head.js') }}"></script>
+  <script type="text/javascript" src="{{ asset_url('js/chart.js') }}"></script>
```
Comment (heyitsmebev) [inline app/templates/old/admin_template.html:27]: we wont need this here

File: app/assets/js/chart.js
Comment (heyitsmebev) [inline app/assets/js/chart.js:1]: Do we need this?

File: app/templates/views/dashboard/dashboard.html [hunk 2]
```diff
@@ -18,20 +18,51 @@ <h1 class="usa-sr-only">Dashboard</h1>
...
+          <option value="service" selected>{{ current_service.name }}</option>
+          <option value="individual">{{ current_user.name }}</option>
+        </select>
+      </form>
+      <div id="activityChart">
+        <div class="chart-header">
+          <div class="chart-subtitle">{{ current_service.name }} - Last 7 Days</div>
+          <div class="chart-legend" aria-label="Legend"></div>
+        </div>
+        <div class="chart-container" id="weeklyChart" data-service-id="{{ service_id }}"></div>
```
Comment (heyitsmebev) [inline app/templates/views/dashboard/dashboard.html:55]: do we need to pass the service_id?

File: app/assets/js/chart.js
Comment (jonathanbobel) [reply app/assets/js/chart.js:1]: we do not, removing

**Review by terrazoon — APPROVED** [review]
I think when I used d3 more than 5 years ago, I just grabbed some google examples for pie charts and bar charts and made them into reusable classes, but for the tables that was just plain html and not part of d3.

I defer to you on this because I see you are making these tables accessible, etc.

File: app/templates/views/dashboard/dashboard.html [hunk 3]
```diff
@@ -98,7 +129,9 @@ <h2 class="margin-top-4 margin-bottom-1">Recent Batches</h2>
       </table>
     </div>
 
-    <h2 class="margin-top-4 margin-bottom-1">Message count</h2>
+    <h2>Recent templates</h2>
+    {{ ajax_block(partials, updates_url, 'template-statistics') }}
+    {#  <h2 class="margin-top-4 margin-bottom-1">Message count</h2>
```
Comment (heyitsmebev) [inline app/templates/views/dashboard/dashboard.html:134]: Same here, could we delete if we are planning to comment out?

File: gulpfile.js [hunk 4]
```diff
@@ -27,6 +27,7 @@ plugins.uglify = require('gulp-uglify');
 const paths = {
   src: 'app/assets/',
   dist: 'app/static/',
+  templates: 'app/templates/',
```
Comment (heyitsmebev) [inline gulpfile.js:30]: you might have a merge conflict here. I believe Alex removed this line

**Review by heyitsmebev — CHANGES_REQUESTED** [review]
Good job with all the javascript and js testing! I can tell a lot of hard work was put into this! I left a couple small feedbacks, otherwise LGTM

File: app/assets/javascripts/activityChart.js [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+        if (!ctx) {
+            return;
+        }
+
+        var socket = io();
+        var eventType = type === 'service' ? 'fetch_daily_stats' : 'fetch_daily_stats_by_user';
+        var socketConnect = type === 'service' ? 'daily_stats_update' : 'daily_stats_by_user_update';
+
+        socket.on('connect', function () {
+            const userId = ctx.getAttribute('data-service-id'); // Assuming user ID is the same as service ID
```
[the remark this answers is not in the document]
Comment (heyitsmebev) [reply app/assets/javascripts/activityChart.js:193]: const userId = ctx.getAttribute('data-service-id');

**Review by heyitsmebev — APPROVED** [review]
LGTM

File: gulpfile.js (hunk 4 above)
Comment (heyitsmebev) [reply gulpfile.js]: I removed the extra line and synced with main in order for this PR to get merged today

**Review by ccostino — APPROVED** [review]
Thank you, @jonathanbobel! 🎉 

(and thank you @heyitsmebev for the quick assist with the final sync with `main`!)

There _might_ be a slight issue with the chart displaying messages over the last 7 days, I'm not seeing any show up despite sending some test ones, but I think it's better to get this foundation in and then fix/tweak what still needs to be done, rather than hold the whole thing up.

**Pull request merged**